tests/ci/test_test_quality_guardrails.py

```
@@ -68,24 +80,39 @@ def _git_changed_test_files() -> list[Path]:
-            text=True,
-            cwd=FO_ROOT,
-        )
-        if not result.stdout.strip():
+    changed: set[str] = set()
+    for diff_args in (
+        ["origin/main...HEAD"],
+        ["--cached"],
+        ["HEAD"],
+    ):
+        try:
             result = subprocess.run(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)